bin/smack.py

```diff
@@ -140,11 +198,22 @@ def arguments():
   verifier_group.add_argument('--smackd', action="store_true", default=False,
     help='generate JSON-format output for SMACKd')
 
+  svcomp_group = parser.add_argument_group('svcomp options')
+
+  svcomp_group.add_argument('--svcomp', action="store_true", default=False,
+    help='enter svcomp mode')
+
+  svcomp_group.add_argument('--syntax-check', action="store_true", default=False,
```

Comment:
Shaobo put this in - I am not sure it does anything yet...

However, I kind of had a different idea.  Instead of having an explicit `--syntax-check` flag that validates generated .bpl files, I think it makes sense to always validate bpl files to see if they are valid boogie programs by passing them through boogie by calling `boogie /noVerify ...`.  When the `--no-verify` flag is not set, either boogie or corral validate the bpl files for parse, name resolution, and typechecking errors.  It seems reasonable that we should still validate bpl files even when they aren't being verified.

We haven't integrated this validation functionality yet because boogie does not set a non-zero exit code when one of these parsing related issues is raised.  There is currently a proposed patch to fix this (boogie-org/boogie#13).  We'll add this functionality when the patch is completed and accepted.